dtfs-central-api/src/v1/controllers/utilisation-report-service/post-add-fees-to-an-existing-payment-group.controller/index.ts

```diff
@@ -27,25 +27,27 @@ export type PostAddFeesToAnExistingPaymentGroupRequest = CustomExpressRequest<{
  */
 export const postAddFeesToAnExistingPaymentGroup = async (req: PostAddFeesToAnExistingPaymentGroupRequest, res: Response) => {
   const { reportId } = req.params;
-  const { feeRecordIds, paymentIds, user } = req.body;
+  const { feeRecordIds, paymentIds: requestedPaymentIds, user } = req.body;
 
   try {
-    const payments = await PaymentRepo.findByIdAndReportIdWithFeeRecordsWithReportAndPayments(paymentIds, Number(reportId));
+    const payments = await PaymentRepo.findPaymentsInGroupContainingPaymentByIdAndReportIdWithFeeRecords(requestedPaymentIds[0], Number(reportId));
     if (payments.length === 0) {
       throw new NotFoundError('No payments found for the given payment IDs.');
     }
 
-    validateThatSelectedPaymentsBelongToSamePaymentGroup(payments);
-    validateThatSelectedPaymentsFormACompletePaymentGroup(payments, paymentIds);
+    validateThatRequestedPaymentsMatchSavedPayments(payments, requestedPaymentIds);
 
-    const existingPaymentGroupFeeRecords = payments.at(0)?.feeRecords;
+    const existingPaymentGroupFeeRecords = payments[0].feeRecords;
     validateThatPaymentGroupHasFeeRecords(existingPaymentGroupFeeRecords);
 
     const selectedFeeRecords = await FeeRecordRepo.findByIdAndReportIdWithReport(feeRecordIds, Number(reportId));
+
+    const paymentGroupCurrency = payments[0].currency;
+    validateThatAllSelectedFeeRecordsAndPaymentGroupHaveSameCurrency(selectedFeeRecords, paymentGroupCurrency);
```

Comment:
Do we have either a unit or api test covering this?

Reply:
Added a unit test for this case, thanks